languageServer/src/TextDocumentWrapper.ts

```diff
@@ -0,0 +1,272 @@
+import {
+  CompletionItem,
+  DocumentSymbol,
+  Position,
+  SymbolKind,
+  Range,
+  Location,
+  InsertTextFormat,
+  CodeActionParams,
+  CodeAction,
+  CodeActionKind,
+  WorkspaceEdit,
+  TextEdit
+} from 'vscode-languageserver/node'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import {
+  isNode,
+  isScalar,
+  parseDocument,
+  visit,
+  Node,
+  Document,
+  Scalar,
+  Pair,
+  isPair
+} from 'yaml'
+import { alphadecimalWords, variableTemplates } from './regexes'
+import { DvcYaml } from './DvcYamlModel'
+import { ITextDocumentWrapper } from './ITextDocumentWrapper'
+
+type CompletionTemplateBody = string | { [key: string]: CompletionTemplateBody }
+interface CompletionTemplate {
+  label: string
+  body: CompletionTemplateBody
+}
+export class TextDocumentWrapper implements ITextDocumentWrapper {
+  uri: string
+
+  private textDocument: TextDocument
+  private pythonFilePaths: string[] = []
+
+  constructor(textDocument: TextDocument, pythonFilePaths?: string[]) {
```

Comment:
Could be `constructor(textDocument: TextDocument, pythonFilePaths: string[] = []) {` instead of `this.pythonFilePaths = pythonFilePaths ?? []`